File: Models/Item/Item.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace sales_order.Items.Models
{
    public class Item
    {
        [Key]
        public int ItemCode { get; set; }

        [Required]
        public string ItemName { get; set; }

        [Required]
        public string description { get; set; }

        [Required]
        public int StockQty { get; set; }

        [Required]
        [Column(TypeName = "decimal(10,2)")]
        public decimal UnitPrice { get; set; }
    }
}

File: Client/src/item/typing/Item.ts
import { IItem } from './IItem';

export class Item {
  itemCode?: number;
  itemName: string;
  description: string;
  stockQty: number;
  unitPrice: number;
  constructor(props: IItem) {
    if (props.itemCode) {
      this.itemCode = props.itemCode;
    }
    this.itemName = props.itemName;
    this.description = props.description;
    this.stockQty = props.stockQty;
    this.unitPrice = props.unitPrice;
  }
}
